src/parser.js

```diff
@@ -386,24 +437,114 @@ export default class Parser extends Stream {
             },
             'skip'() {
               this.manifest.skip = entry.attributes;
+
+              if (!entry.attributes.hasOwnProperty('SKIPPED-SEGMENTS')) {
+                this.trigger('warn', {
+                  message: '#EXT-X-SKIP lacks required attribute SKIPPED-SEGMENTS'
+                });
+              }
             },
             'part'() {
               this.manifest.parts = this.manifest.parts || [];
               this.manifest.parts.push(entry.attributes);
+              const missingAttributes = [];
+
+              ['URI', 'DURATION'].forEach(function(k) {
+                if (!entry.attributes[k]) {
+                  missingAttributes.push(k);
+                }
+              });
+
+              if (missingAttributes.length) {
+                const index = this.manifest.parts.length - 1;
+
+                this.trigger('warn', {
+                  message: `#EXT-X-PART #${index} lacks required attribute(s) ${missingAttributes.join(', ')}`
+                });
+              }
+
+              if (this.manifest.renditionReports) {
+                this.manifest.renditionReports.forEach((r, i) => {
+                  if (!r.hasOwnProperty('LAST-PART')) {
+                    this.trigger('warn', {
+                      message: `#EXT-X-RENDITION-REPORT #${i} lacks required attribute(s) LAST-PART`
```

Comment:
so, this is only required if the manifest has a partial segment. Can we inspect here to see if we got one or not? I can imagine that it's possible to have a rendition report without partial segments.

Reply:
This is within the part handler, just incase the first rendition report happens before the first part.